qutebrowser/app.py

```diff
@@ -343,15 +343,12 @@ def on_focus_changed(_old, new):
         log.misc.debug("on_focus_changed called with non-QWidget {!r}".format(
             new))
 
-    if new is None or not isinstance(new, mainwindow.MainWindow):
-        try:
-            objreg.delete('last-focused-main-window')
-        except KeyError:
-            pass
-        qApp.restoreOverrideCursor()
-    else:
-        objreg.register('last-focused-main-window', new.window(), update=True)
+    if new is not None:
```

Comment:
> You removed the isinstance check here, which means the mouse cursor would remain hidden when switching focus to a dialog box or something similar.

Re-including this check would break `_maybe_hide_mouse_cursor()` again - because the issue is that a `MainWindow` never gets directly focused (at least on my end), it's always some child - in this case the webkit-specific WebView.

So unless we have some good way of distinguishing between if what I clicked was an object that should cause the cursor to be re-shown or not, I don't see a way to reimplement this feature in a way that actually works. (Since the current almost surely doesn't)